support/ci/promote_rc.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+set -e
+
+if [ -z ${HAB_AUTH_TOKEN+x} ]; then
+  echo "HAB_AUTH_TOKEN var is unset"
+  exit 1
+fi
+if [ -z ${BINTRAY_USER+x} ]; then
+  echo "BINTRAY_USER var is unset"
+  exit 1
+fi
+if [ -z ${BINTRAY_KEY+x} ]; then
+  echo "BINTRAY_KEY var is unset"
+  exit 1
+fi
+
+HAB_VERSION=$(cat VERSION)
+PROMOTE_CHANNEL=stable
+RC_CHANNEL=rc-$HAB_VERSION
+RC_PKGS_JSON=$(curl https://bldr.habitat.sh/v1/depot/channels/core/$RC_CHANNEL/pkgs)
+RC_PKGS_JSON_DATA=$(echo $RC_PKGS_JSON | jq -c -r '.data | .[]')
+TOTAL_COUNT=$(echo $RC_PKGS_JSON | jq -r '.total_count')
+END=$(echo $RC_PKGS_JSON | jq -r '.range_end')
+
+if [ $(($END+1)) !=  $TOTAL_COUNT ]; then
+  echo "There are multipe pages of releases. Consider deleting the channel and then rebuild."
+  exit 1
+fi
+
+declare -A pkg_releases
+for pkg in $RC_PKGS_JSON_DATA
+do
+    ORIGIN=$(echo $pkg | jq -r ".origin")
+    NAME=$(echo $pkg | jq -r ".name")
+    VERSION=$(echo $pkg | jq -r ".version")
+    RELEASE=$(echo $pkg | jq -r ".release")
+    TARGET=$(echo $pkg | jq -r ".platforms[0]")
```

Comment:
As an aside... why is `platforms` an array here (and not a scalar `platform`) if we only build releases for a single platform?

Reply:
yeah that's a great question. I just dug into that service code and it looks like we use some generic logic to handle listing packages. Such lists can be as granular as listing info by release (as we do here) or might list info by version or even origin/name. At the release level, it is extremely unlikely if not impossible there would be more than a single platform but there may be multiple platforms at the version or name level.